### PR TITLE
test(express-tooling): use spies to test whether function is called (#125)

### LINKED ISSUE #125: Use spies to test whether function is called
Replace such code: https://github.com/zalando-incubator/authmosphere/blob/67316446415fdad1a37e12ba9d745bf7375fdc28/test/unit/express-tooling.ts#L47 via using something like http://chaijs.com/plugins/chai-spies/

### PR DESCRIPTION
**Pull request by hypescript**
close #125

### CODE REVIEW

**Comment by Retro64** [comment]
Looks good to me, but we have to clarify how to proceed in connection to the 2.0 branch - therefore no :+1: (and merge) yet

File: test/unit/express-tooling.ts [hunk 1]
```diff
@@ -63,16 +65,12 @@ describe('express tooling', () => {
...
-        const next = () => {
-          called = true;
-        };
 
         // when
         requireScopesMiddleware(requiredScopes)(requestMock, responseMock, next);
 
         // then
-        return expect(called).to.be.false;
+        return expect(next).to.not.have.been.called; // async calls won't be detected
```
Comment (bzums) [inline test/unit/express-tooling.ts]: I do not exactly understand the comment here, can you elaborate on that?

Comment (hypescript) [reply test/unit/express-tooling.ts]: What I mean is, if there is an asynchronous call to next(), e.g. something like
```javascript
Promise.resolve().then(next)
```
somewhere inside the middleware call, it won't be detected, just as it wouldn't be detected in previous implementation of the test.

Comment (bzums) [reply test/unit/express-tooling.ts]: I see, thanks. So to "fix" this we would have to return the promise from `requireScopesMiddleware`, maybe that is a good idea anyway...

Comment (hypescript) [reply test/unit/express-tooling.ts]: It might be a good idea, just beware of unhandled promise rejections.

**Comment by bzums** [comment]
@hypescript we created a branch for version 2.0.0 in which we want to merge a number of breaking changes, including this PR if possible. Can you create a new PR against https://github.com/zalando-incubator/authmosphere/tree/authmosphere-2-0-0? That would be great, thanks :) (afterwards we would close this PR)

File: test/unit/express-tooling.ts (hunk 1 above)
Comment (bzums) [reply test/unit/express-tooling.ts]: FYI: we created a follow up ticket for this issue #132

**Comment by bzums** [comment]
👍

**Comment by Retro64** [comment]
:+1:

**Comment by Retro64** [comment]
@hypescript will be released with 2.0.0

**Pull request merged**